vespa/deployment.py

```diff
@@ -874,8 +1092,44 @@ def get_dev_region(self) -> str:
         return self._request("GET", "/zone/v1/environment/dev/default")["name"]
 
     def get_prod_region(self):
-        # TODO Support multiple regions
-        return self.application_package.deployment_config.regions[0]
+        regions = self.get_prod_regions()
+        return regions[0]
+
+    def get_regions_from_deployment_xml(
+        self, disk_folder: Optional[str] = None
+    ) -> List[str]:
+        # Parse the XML data from file
+        if disk_folder is None:
+            if self.disk_folder is None:
+                disk_folder = os.path.join(os.getcwd(), self.application)
+            else:
+                disk_folder = self.disk_folder
+        deployment_xml_path = Path(disk_folder) / "deployment.xml"
+        try:
+            with open(deployment_xml_path) as f:
+                xml_data = f.read()
+        except FileNotFoundError:
+            raise FileNotFoundError(
+                f"deployment.xml not found in {deployment_xml_path}. Provide disk_folder with the path to application root."
+            )
+        root = ET.fromstring(xml_data)
+
+        # Find all 'region' elements that have 'prod' as a parent
+        regions = [
+            region.text
+            for prod in root.findall(".//prod")
+            for region in prod.findall("region")
```

Comment:
Regions aren't necessarily direct children of `prod`, but it _is_ true that a `region` will always have exactly one `prod` ancestor, and all `region` descendants of a `prod` are prod-deployment regions.

Reply:
There is no concept of `instance` here, btw.; perhaps `pyvespa` doesn't support that?

Reply:
Used API for this instead, and added instance-parameter.